bluebird-promisell.js

```diff
@@ -0,0 +1,49 @@
+/* jshint esnext:true */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const Promise = require('bluebird');
+const P = require('bluebird-promisell');
+const R = require('ramda');
+const S = require('sanctuary');
+
+//    readFile :: Object -> String -> Promise Error String
+const readFile = S.flip(Promise.promisify(fs.readFile));
+
+//    readFileWithDirAndName :: String -> String -> Promise Error String
+const readFileWithDirAndName = R.curryN(2, (dir, name) => {
+  const filePath = path.join(dir, name);
+  return readFile({ encoding: 'utf8' })(filePath);
+});
+
+//    readFilesWithDirAndNames :: String -> [String] -> Promise Error [String]
+const readFilesWithDirAndNames = R.curryN(2, (dir, fileNames) => {
+  const readOneFile = readFileWithDirAndName(dir);
+  return P.traversep(readOneFile)(fileNames);
```

Comment:
It's a pity that with promises we're forced to use specialized functions such as `traversep` rather than generalized functions such as [`traverse`][1].


[1]: http://ramdajs.com/docs/#traverse